src/classes/groundplaces.ts

```diff
@@ -59,19 +59,20 @@ export class GroundPlacesController {
    * @returns {GroundPlace[]}
    */
   // @ts-ignore
-  public autocomplete(query: string, filters?: AutoCompleteFilters[]): GroundPlace[] {
+  public autocomplete(query: string, filters?: AutocompleteFilter[]): GroundPlace[] {
     const groundPlaces: GroundPlace[] = this.getGroundPlaces();
 
     // Method toLowerCase() is used because the includes() method is case sensitive
     const currentQuery: string = query.toLowerCase();
 
-    const isFilterByStopGroupActive: boolean = filters?.includes(AutoCompleteFilters.STOP_GROUP);
-    const isFilterByStopClusterActive: boolean = filters?.includes(AutoCompleteFilters.STOP_CLUSTER);
-    const isFilterByServicedActive: boolean = filters?.includes(AutoCompleteFilters.SERVICED);
-    const isFilterWithChildrenActive: boolean = filters?.includes(AutoCompleteFilters.SEGMENT_PROVIDER_STOP);
+    const isFilterByStopGroupActive: boolean = filters?.includes(AutocompleteFilter.STOP_GROUP);
+    const isFilterByStopClusterActive: boolean = filters?.includes(AutocompleteFilter.STOP_CLUSTER);
+    const isFilterByServicedActive: boolean = filters?.includes(AutocompleteFilter.SERVICED);
+    const isFilterWithChildrenActive: boolean = filters?.includes(AutocompleteFilter.SEGMENT_PROVIDER_STOP);
 
     return groundPlaces.filter(
       (place: GroundPlace): GroundPlace => {
+        // Checking that the place matching the search query
         if (
           !(place.type === GroundPlaceType.CLUSTER && place.unique_name.toLowerCase().includes(currentQuery)) &&
```

Comment:
This one is more expensive than the type & serviced filter.
Therefor it should go after ;)

Reply:
These conditions are used for the search query not filter :) 
But you're right, I'll handle this search on the `unique_name` at last :)